prototyping/xarray/xdose.py

```diff
@@ -0,0 +1,198 @@
+from typing import Tuple
+
+import numpy as np
+import pydicom
+import xarray as xr
+
+from pymedphys._dicom import constants, coords, dose
+
+XARRAY_DIMS_BY_COORD_SYSTEM_AND_ORIENT = {
+    "S": ["y", "z", "x"],
+    "P": ["y", "z", "x"],
+    "P dec": ["y", "x", "z"],
+    "D": ["z", "y", "x"],
+    "D dec": ["z", "x", "y"],
+}
+
+
+def xdose_from_dataset(ds, coord_system="SUPPORT", decimals_to_round_coords=-1):
```

Comment:
I'm not convinced that this function should handle rounding at all. That seems like something the user can do with the object later if they desire...